src/main/java/com/verily/genomewarp/utils/GenomeRangeUtils.java

```diff
@@ -261,7 +261,7 @@ else if (vcfRegion.getEnd() <= queryRegion.getStart()) {
     for (GenomeRange currRange: toMerge) {
       if (maxRange != null) {
         if (currRange.getChromosome().equals(maxRange.getChromosome())
-            && currRange.getStart() < maxRange.getStart()) {
+            && currRange.compareTo(maxRange) < 0) {
           GenomeWarpUtils.fail(logger, "Input regions are not sorted by pos");
```

Comment:
I agree that the message could be much more clear but this condition will only occur if there is a logic error somewhere. It doesn't indicate the problem with the user's inputs, we run this method on sorted list so it is just a sanity check that should never occur to the end user (and even if it will, it means that there is a bug, it's not something that can be fixed without looking at the code).
So I don't think we should expand on this message, for debugging purposes it's enough and it wouldn't help users.
If you disagree or have a proposal of a better wording, I would happily change this (and potentially others) fail message.

Reply:
Hi Ekaterina,

That sounds fine for now, but looking into the future when you have more complex programs with many layers, you will want to be dealing less with code and more with composing them.  If these fundamental building blocks are well-bounded in your error-space, then you shouldn't need to look at the source again.

Regarding errors, in most programming languages you will see them in all lower-case.  Here's an example from Java:

```Java
paul$ javac hw.java
hw.java:4: error: <identifier> expected
    public statc void main(String[] args) {
                ^
hw.java:4: error: invalid method declaration; return type required
    public statc void main(String[] args) {
                      ^
2 errors
paul$ javac hw.java
hw.java:6: error: cannot find symbol
        System.ot.println("Hello, World");
              ^
  symbol:   variable ot
  location: class System
1 error
```

Hope it helps,
~p